tensorflow/python/keras/layers/recurrent.py

```diff
@@ -82,7 +82,7 @@ def __init__(self, cells, **kwargs):
       if not hasattr(cell, 'call'):
         raise ValueError('All cells must have a `call` method. '
                          'received cells:', cells)
-      if not hasattr(cell, 'state_size'):
```

Comment:
I see, if the state_size has some side effects, we probably should just use dir() for it. Also with dir(), we probably can skip the check for hasattr(). Also for consistency, please update the check for call() as well. Thanks.